### PR TITLE
Add thorchain reporting

### PR DESCRIPTION
**Pull request by paullinator**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202935969322502

### CODE REVIEW

File: src/partners/thorchain.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+      // and the transaction failed
+      if (match != null) {
+        status = 'failed'
+      }
+
+      const timestampMs = div(date, '1000000', 16)
+      const { timestamp, isoDate } = smartIsoDateFromTimestamp(timestampMs)
+
+      const [chainAsset] = srcAsset.split('-')
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (samholmes) [inline src/partners/thorchain.ts:123]: Why is this needed if `_` is used for the unused var? Perhaps try `_` alone instead of `_chain`?

File: src/partners/thorchain.ts [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+const asThorchainResult = asObject({
+  actions: asArray(asUnknown)
+})
+type Settings = ReturnType<typeof asSettings>
+type ThorchainResult = ReturnType<typeof asThorchainResult>
+
+const ROLLBACK = 500
+const LIMIT = 50
+const THORCHAIN_MULTIPLIER = 100000000
```
Comment (samholmes) [inline src/partners/thorchain.ts:55]: RUNE_MULTIPLIER?

File: src/cacheEngine.ts [hunk 3]
```diff
@@ -91,113 +91,121 @@ export async function cacheEngine(): Promise<void> {
...
-    const monthStart = startOfMonth(new Date(Date.now()))
-    start = sub(monthStart, { months: 1 }).getTime() / 1000
-  } catch {
-    start = new Date(2017, 1, 20).getTime() / 1000
-  }
+    try {
+      await reportsMonth.get('initialized:initialized')
+      const monthStart = startOfMonth(new Date(Date.now()))
+      start = sub(monthStart, { months: 1 }).getTime() / 1000
+    } catch (e) {
```
Comment (samholmes) [inline src/cacheEngine.ts:102]: `(e: any)`?

**Review by samholmes — APPROVED** [review]

File: src/partners/thorchain.ts (hunk 2 above)
Comment (paullinator) [reply src/partners/thorchain.ts:55]: It's entirely protocol wide on Thorchain, not specific to RUNE

File: src/partners/thorchain.ts (hunk 1 above)
Comment (paullinator) [reply src/partners/thorchain.ts:123]: Neither works. Not sure what lint rule needs to change
